plugins/ssl/cert_validity

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# check the certificate validity of your certfificates
+#
+# andreas perhab - andreas.perhab@wt-io-it.at
+# https://www.wt-io-it.at/
+#
+# License: GPLv2
+#
+# Installing: Add list of your certificates prefixed by the type in munin plugin-conf.d
+#
+# For openvpn ca.crt and crl.pem
+# [cert_validity]
+# user root
+# env.CERTS crl:/etc/openvpn/easy-rsa/keys/crl.pem x509:/etc/openvpn/easy-rsa/keys/ca.crt
+#
+# For letsencrypt certficates
+# [cert_validity]
+# user root
+# env.CERTS x509:/etc/letsencrypt/live/domain1.example.com/cert.pem x509:/etc/letsencrypt/live/domain2.example.com/cert.pem
+#
+# Dependencies: openssl
+#
+# Parameters:
+#
+#  config
+
+
+if [ "$1" == "config" ] ; then
+	echo "graph_title Certificate validity"
+	echo "graph_args --logarithmic --base 1000"
+	echo "graph_vlabel certificate validity in days"
+	echo "graph_category certificates"
+fi
+
+now=$(date +%s)
+for cert in ${CERTS[@]}; do
+	cert_type=${cert%:*}
+	cert_file=${cert#*:}
+	cert_name=${cert_file//\//_}
+	cert_name=${cert_name//./_}
+	cert_name=${cert_name//-/_}
+	cert_name=${cert_name#_}
+	if [ "$1" == "config" ] ; then
+		echo "${cert_name}.label ${cert_file}"
+		echo "${cert_name}.warning 5:"
+		echo "${cert_name}.critical 1:"
+	elif [ "$1" == "" ] ; then
+		validity=$(/usr/bin/openssl $cert_type -text -noout -in "$cert_file" | grep -E '(Next Update|Not After)')
+		validity=${validity#*:}
+		validity=$(date --date="$validity" +%s)
+		validity=`expr $validity - $now`
+		if which bc > /dev/null; then
```

Comment:
Maybe just use `awk` for float calculations: `echo "$validity" | awk '{ print($1 / 86400) }'`?

(munin plugins may assume, that `awk` is available)

Reply:
thanks for the suggestion, now the calculation is done with awk